src/audio/pcm_converter/pcm_converter.c

```diff
@@ -33,9 +33,9 @@ void pcm_convert_as_linear(const struct audio_stream *source, uint32_t ioffset,
 
 	while (i < samples) {
 		/* calculate chunk size */
-		N1 = audio_stream_bytes_without_wrap(source, r_ptr);
-		N2 = audio_stream_bytes_without_wrap(sink, w_ptr);
-		chunk = MIN(N1, N2) * s_size_out;
+		N1 = audio_stream_bytes_without_wrap(source, r_ptr) / s_size_in;
```

Comment:
what is s_size_in and out ? if it's the format size then this is always a power of two and we can >> instead of doing expensive divsions.

Reply:
Updated

Reply:
We should minimise this and use shift as our sample size is either 2 or 4 bytes and will result in a performance bump.